src/UCTNode.cpp

```diff
@@ -220,32 +221,43 @@ UCTNode* UCTNode::uct_select_child(int color) {
     auto parentvisits = size_t{0};
     for (const auto& child : m_children) {
         if (child->valid()) {
-            parentvisits += child->get_visits();
-            if (child->get_visits() > 0) {
+            if (!child->first_visit()) {
+                parentvisits += child->get_visits();
                 total_visited_policy += child->get_score();
+            } else if (child->get_score() > best_unvisited_psa) {
+                best_unvisited_psa = child->get_score();
+                best = child.get();
             }
         }
     }
 
+    if (parentvisits == 0) {
+        assert(best != nullptr);
+        return best;
+    }
+
     auto numerator = std::sqrt((double)parentvisits);
-    auto fpu_reduction = cfg_fpu_reduction * std::sqrt(total_visited_policy);
-    // Estimated eval for unknown nodes = original parent NN eval - reduction
-    auto fpu_eval = get_net_eval(color) - fpu_reduction;
+    auto best_value = -DBL_MAX;
```

Comment:
```std::numeric_limits<double>::min()```

Reply:
Ok but I think min is the smallest positive number?
http://en.cppreference.com/w/cpp/types/numeric_limits
So lowest should be used instead.

Reply:
Ok but I think min is the smallest positive number?
http://en.cppreference.com/w/cpp/types/numeric_limits
So lowest should be used instead.

Reply:
You're right (but the difference probably doesn't matter for this code?).